src/test/java/mops/presentation/DirectoryControllerTest.java

```diff
@@ -195,6 +195,24 @@ void editFolder() throws Exception {
                         )));
     }
 
+    /**
+     * Tests the route after renaming a folder.
+     */
+    @Test
+    @WithMockKeycloackAuth(roles = "studentin", idToken = @WithIDToken(email = "user@mail.de"))
+    void renameFolder() throws Exception {
+        mockMvc().perform(post("/material1/dir/{dirId}/rename", 1)
+                .param("originDirId", String.valueOf(1L))
+                .param("newName", "New Name")
+                .with(csrf()))
+                .andExpect(status().is3xxRedirection())
+                .andExpect(redirectedUrl("/material1/dir/1"))
+                .andDo(document("index/DirectoryController/{method-name}",
+                        pathParameters(
+                                parameterWithName("dirId").description("The directory id.")
+                        )));
```

Comment:
Maybe add a verify of directoryService.renameFolder()?

Reply:
Those are the controller tests, we never checked the services, just the http response 🤷‍♀